nilearn/datasets/atlas.py

```diff
@@ -2047,7 +2391,24 @@ def fetch_atlas_pauli_2017(
 
     fdescr = get_dataset_descr(dataset_name)
 
-    return Bunch(maps=atlas_file, labels=labels, description=fdescr)
+    if atlas_type == "probabilistic":
+        return Bunch(
+            maps=atlas_file,
+            labels=labels,
+            description=fdescr,
+            atlas_type=atlas_type,
+        )
+
+    lut = _generate_atlas_look_up_table("fetch_atlas_pauli_2017", name=labels)
+    _check_look_up_table(lut=lut, atlas=atlas_file)
+
+    return Bunch(
+        maps=atlas_file,
+        labels=labels,
+        description=fdescr,
+        lut=lut,
+        atlas_type=atlas_type,
+    )
```

Comment:
Actually this code repeats too much. Could it be a function? That would increase readability.

Reply:
Yes, that's what I mean, including the if condition actually.